### ISSUE TITLE
Way to easily disable certain biomes[Enhancement]

### ISSUE THREAD

**Issue by McCheadle** [Stardust-Labs-MC/Terralith#103]
### Describe your enhancement/improvement, with context and reasoning.

I absolutely love this mod but I don't like it when there are floating islands in my world. That's why I would want to be able to disable all of the skyland biomes. Since I didn't find a config file, I looked online for some anwers but the solutions were all a bit too technical for me. If there is an easy way to do this then please tell me. It would make me enjoy the mod even more than I do now. Being able to easily disable any biome you want would also just be an amazing feature. 
Thank you for the amazing mod.

**Comment by catter1** [comment]
In our Discord, we have these instructions for removing a biome. I don't know if this is the "too technical" version that you found online.
![image](https://github.com/Stardust-Labs-MC/Terralith/assets/63272345/79f16822-4779-42d9-bd1d-c8399cb5bb37)

For an easier option, you can try using mods like [Biome Replacer](https://modrinth.com/mod/biome-replacer).

But to directly to answer your question: the reason Terralith doesn't have a config is that Terralith is a datapack, and datapacks cannot have configs. (Even as a mod - Terralith is just a datapack with `.jar` at the end.) Whilst it is technically possible for us to "mod" it enough to give Terralith some sort of config, it will take a lot of work and time that we won't have for a while. And even if we do end up ever adding a config, I'm not sure if disabling biomes will be an option. It's very complicated without *completely* modding Terralith's biome distribution, which we are not going to do.

**Comment by McCheadle** [comment]
thanks for the response. And keep up the good work.

**Comment by matzar24** [comment]
> In our Discord, we have these instructions for removing a biome. I don't know if this is the "too technical" version that you found online. ![image](https://github.com/Stardust-Labs-MC/Terralith/assets/63272345/79f16822-4779-42d9-bd1d-c8399cb5bb37)
> 
> For an easier option, you can try using mods like [Biome Replacer](https://modrinth.com/mod/biome-replacer).
> 
> But to directly to answer your question: the reason Terralith doesn't have a config is that Terralith is a datapack, and datapacks cannot have configs. (Even as a mod - Terralith is just a datapack with `.jar` at the end.) Whilst it is technically possible for us to "mod" it enough to give Terralith some sort of config, it will take a lot of work and time that we won't have for a while. And even if we do end up ever adding a config, I'm not sure if disabling biomes will be an option. It's very complicated without _completely_ modding Terralith's biome distribution, which we are not going to do.
-------------------------------------------------------------------
Are there weight values for biomes? Or size value?
What are the methodes for changing biome size? Mod, or ?

**Comment by catter1** [comment]
There are no weights. This is the guide we have for configuring Terralith's biome sizes. There is not an easy mod to use for this that I am aware of.

![Image](https://github.com/user-attachments/assets/17b2f688-0c6a-48ab-b534-fbb33f790e2b)

**Comment by matzar24** [comment]
> There are no weights. This is the guide we have for configuring Terralith's biome sizes. There is not an easy mod to use for this that I am aware of.
> 
> ![Image](https://github.com/user-attachments/assets/17b2f688-0c6a-48ab-b534-fbb33f790e2b)
---------------------------------------------------------------------------------------------------------------
From where are these screeshots? As I see now - I dont see any wiki-subtitle for configuration
I used Biome Sizes mod, I think biomes are larger, but I dont know how it affects terrain generation?
Comparably, what is better, xz_scale or Biome Sizes mod ?
Or they are just using the same thing to change biome size?
Does terrain retain its proportions? I only want change size of biome, dont want to touch terrain shape size.

Another thing, Recomended map.jacobsjo.eu map viewer doesnt show the same thing
Loaded Tectonic + terralith same seed 1.20.1 and see different biomes
From tests I think it doesnt count in Tectonic

**Comment by catter1** [comment]
These screenshots are from our discord. I do not know anything about Biome Sizes mod, so you should redirect your questions to them.

map.jacobsjo.eu is accurate. However, when using datapacks, Terralith and Tectonic are **not** compatible. You need to use Terratonic instead of Tectonic. For example, [this link](https://map.jacobsjo.eu/?datapacks=modrinth:terralith,modrinth:terratonic) uses the correct combination.

**Comment by matzar24** [comment]
You mean incompatible ingame, or only in map viewer?
I'm in world on mods Terralith + Tectonic, and it looks like it works, but I dont know how it's supposedly look originally so I cannot judge if they're compatible. I mean, if they looks different on map viewer than in my world, then something works, but what exactly you meant by saying Terralith + Tectonic are incompatible?

Also saw this:
"If you are using the mod version of Terralith, you do not need this! Use the Tectonic mod, as it has Terratonic built in."
So I gues you meant to use this datapack only for map viewing? Am I right?
Or In second scenario if I would use Terralith as datapack, then I would need use terratonic too. Yes?

**Comment by catter1** [comment]
I mean exactly the part you say in the second part. **As datapacks**, Terralith and Tectonic are incompatible. The mod versions work great together. But when you're using datapacks, you must use Terratonic instead of Tectonic. Both in game and in the map viewer.

**Comment by matzar24** [comment]
It would be so amazing if there would be feature including map viewer setting for xz_scale, or if values are edited in uploaded file  it'll work?
I uploaded edited version with xz_scale=0.5, so x4 smaller biomes, but they're looking like on original scale. ?

**Comment by catter1** [comment]
The map viewer doesn't have any issues displaying this - after all, those "edits" are just how datapacks work. If you still have Tectonic/Terratonic on there, that's why - it overrides that. Both in game and in the viewer.

**Comment by matzar24** [comment]
I didnt exactly understood this response, 
you meant I'm able to edit but need clearup cookies or 
terratonic overrides xz_scale, so it wont work?

Loading terralith second, after terratonic makes map viewer not displaying map, grey screen.
Tried in incognito tab
I checked in file and vegetation, temperature, base_erosion are set correctly to xz_scale: 0.5

**Comment by catter1** [comment]
Ignoring the map viewer: Terratonic overrides any changes you make to Terralith's biome size via datapack edits. It is not possible to edit biome sizes in Terralith if you have Terratonic installed.

**Comment by matzar24** [comment]
I see in terratonic is only temperature, two others are absent.
Edited only temperature and no change

**Comment by catter1** [comment]
Correct. Terratonic handles biome sizes differently than Terralith. I do not know how to edit them in Terratonic - simply that it is not possible by editing Terralith itself when Terratonic is installed.

**Comment by matzar24** [comment]
I changed erosion's sx_scale in "tectonic" folder and it worked, still no file for vegeatation.
Biomes in map viewer are larger, but it looks like badlands ate most of dessert.
And changing vegetation in terralith wont change it, dont see any change in map.
I created new Issue on tectonic github, only reporting here my findings.
If you'll find helpful tip, I'll very much be thankful

Vegetation applies to terrain generation  or  trees & plants ?

Finished 15h generation of Terralith+Tectonic 6k radius
(If tectonic overrides xz_scale, so it would be normal .25)
seed -3845798408893511820
![Image](https://github.com/user-attachments/assets/99cf5ca9-c26d-40de-8ccc-127ce50a855b)

Here are comparisons from map viewer
normal .25 vs BigBiomes .125
![Image](https://github.com/user-attachments/assets/20068a31-e873-48c3-bdc9-47c2abd5103e)

![Image](https://github.com/user-attachments/assets/dd88fec5-e706-4772-b696-7c8446538062)

Does xz_scale change % of biomes in world? Like certain biomes will take over others?

Also what are rules for replacing biomes? Sometimes you get crashes when replacing certain biomes, which arent that different in terms of terrain variation. (yes, I have naming and signs correctly replaced : " , ...)
Edit: I just got the hunch, it could be bc I'm replacing terralith biome in vanilla biome, without changing terralith: -> minecraft:
Thats probably why...

Still, are there rules for replacing biomes? Like mountains to dessert, or sth else which is questionable?
You said somewhere to replace skylands to ocean, I guess its bc of they spawn above oceans, but what would happen if replace them to land biome?

One thing I noticed about normal Terralith+Terratonic, there's not a lot deserts, so it would be kinda hard for sand dunes to form :(
Edit: I just compared real generated map vs map viewer, and I see most of badlands's area is covered in sand. ?
